mne/time_frequency/tfr.py

```diff
@@ -209,12 +206,12 @@ def _cwt(X, Ws, mode="same", decim=1, use_fft=True):
     if use_fft:
         fft_Ws = np.empty((n_freqs, fsize), dtype=np.complex128)
     for i, W in enumerate(Ws):
-        if len(W) > n_times:
-            raise ValueError('At least one of the wavelets is longer than the '
-                             'signal. Use a longer signal or shorter '
-                             'wavelets.')
         if use_fft:
             fft_Ws[i] = fft(W, fsize)
+        elif mode == 'same' and len(W) > n_times:
+            warn("At least one of the wavelets is longer than the "
+                 "signal. Switching to 'full' convolution mode.")
```

Comment:
So for now, `warn` in cases where `len(W) > n_times` when `use_fft=True`, and throw an error (or fix the implementation) when `use_fft=False`?